src/java.base/share/classes/java/lang/classfile/Signature.java

```diff
@@ -222,12 +243,20 @@ public static ClassTypeSig of(String className, TypeArg... typeArgs) {
          * {@return a class type signature}
          *
          * @param outerType signature of the outer type, may be {@code null}
+         *        to indicate this is a top-level class or interface
          * @param className the name of this class or interface
          * @param typeArgs the type arguments
+         * @throws IllegalArgumentException if {@code className} cannot be
+         *         {@linkplain Signature##identifier denoted}, such as due to
+         *         the presence of ASCII characters {@code < > :}
          */
         public static ClassTypeSig of(ClassTypeSig outerType, String className, TypeArg... typeArgs) {
-            requireNonNull(className);
-            return new SignaturesImpl.ClassTypeSigImpl(Optional.ofNullable(outerType), className.replace(".", "/"), List.of(typeArgs));
```

Comment:
We should declare that the method accepts binary class name only If we remove conversion from binary to internal class name.